src/sonic_ax_impl/mibs/ietf/rfc3433.py

```diff
@@ -0,0 +1,402 @@
+    """
+    Transceiver tx bias sensor.
+    (TYPE, SCALE, PRECISION) set according to SFF-8472
+    Sensor measures in range (0 mA, 131 mA) with step 2E-3 mA
+    """
+
+    TYPE = EntitySensorDataType.AMPERES
+    SCALE = EntitySensorDataScale.MILLI
+    PRECISION = 3
+
+
+# mapping between DB key and Sensor object
+TRANSCEIVER_SENSOR_MAP = {
+    "temperature": XcvrTempSensor,
+    "voltage":     XcvrVoltageSensor,
+    "rx1power":    XcvrRxPowerSensor,
+    "rx2power":    XcvrRxPowerSensor,
+    "rx3power":    XcvrRxPowerSensor,
+    "rx4power":    XcvrRxPowerSensor,
+    "tx1bias":     XcvrTxBiasSensor,
+    "tx2bias":     XcvrTxBiasSensor,
+    "tx3bias":     XcvrTxBiasSensor,
+    "tx4bias":     XcvrTxBiasSensor,
+}
+
+
+def get_transceiver_sensor(sensor_key):
+    """
+    Gets transceiver sensor object
+    :param sensor_key: Sensor key from XcvrDomDB
+    :param ifindex: Interface index associated with transceiver
+    :return: Sensor object.
+    """
+
+    return TRANSCEIVER_SENSOR_MAP[sensor_key]
+
+
+class PhysicalSensorTableMIBUpdater(MIBUpdater):
+    """
+    Updater for sensors.
+    """
+
+    def __init__(self):
+        """
+        ctor
+        """
+
+        super().__init__()
+
+        self.statedb = SonicV2Connector()
+        self.statedb.connect(self.statedb.STATE_DB)
+
+        # list of available sub OIDs
+        self.sub_ids = []
+
+        # sensor MIB requiered values
+        self.ent_phy_sensor_type_map = {}
+        self.ent_phy_sensor_scale_map = {}
+        self.ent_phy_sensor_precision_map = {}
+        self.ent_phy_sensor_value_map = {}
+        self.ent_phy_sensor_oper_state_map = {}
+
+    def reinit_data(self):
+        """
+        Reinit data, clear cache
+        """
+
+        # clear cache
+        self.ent_phy_sensor_type_map = {}
+        self.ent_phy_sensor_scale_map = {}
+        self.ent_phy_sensor_precision_map = {}
+        self.ent_phy_sensor_value_map = {}
+        self.ent_phy_sensor_oper_state_map = {}
+
+        transceiver_dom_keys = mibs.transceiver_dom_table("*")
+        self.transceiver_dom = self.statedb.keys(self.statedb.STATE_DB,
+                                                 transceiver_dom_keys)
+
+    def update_data(self):
+        """
+        Update sensors cache.
+        """
+
+        self.sub_ids = []
+
+        if self.transceiver_dom is None:
+            return
+
+        # decode keys from DB
+        transceiver_dom = map(lambda k: k.decode("utf-8"), self.transceiver_dom)
+        # update transceiver sensors cache
+        for transceiver_dom_entry in transceiver_dom:
+            # extract interface name
+            interface = transceiver_dom_entry.split("|")[-1]
+            ifindex = port_util.get_index_from_str(interface)
+
+            if ifindex is None:
+                continue
+
+            # get transceiver sensors from transceiver dom entry in STATE DB
+            transceiver_dom = self.statedb.get_all(self.statedb.STATE_DB,
```

Comment:
>transceiver_dom [](start = 12, length = 15)

why you change loop range inside a loop?

Reply:
Sorry for that bad var naming. Agree this is confusing. Actually didn't noticed it because python allows have the same variable name in the inner scope without overriding outer scope variable. Now i fixed it